metricflow/api/simple_api.py

```diff
@@ -0,0 +1,259 @@
+from typing import Dict, List, Optional
+
+from metricflow.configuration.config_handler import ConfigHandler
+from metricflow.configuration.constants import CONFIG_DWH_SCHEMA
+from metricflow.dataflow.sql_table import SqlTable
+from metricflow.engine.metricflow_engine import (
+    MetricFlowEngine,
+    MetricFlowExplainResult,
+    MetricFlowQueryRequest,
+    MetricFlowQueryResult,
+)
+from metricflow.engine.models import Dimension, Materialization, Metric
+from metricflow.engine.utils import build_user_configured_model_from_config, convert_to_datetime
+from metricflow.model.model_validator import ModelValidator
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.semantic_model import SemanticModel
+from metricflow.model.validations.validator_helpers import ValidationIssueType
+from metricflow.protocols.sql_client import SqlClient
+from metricflow.sql.optimizer.optimization_levels import SqlQueryOptimizationLevel
+from metricflow.sql_clients.sql_utils import make_sql_client_from_config
+
+
+class MetricFlowSimpleAPI:
+    """Simple MetricFlow Python Interface."""
+
+    def __init__(
+        self,
+        sql_client: Optional[SqlClient] = None,
+        semantic_model: Optional[SemanticModel] = None,
+        system_schema: Optional[str] = None,
+    ):
+        """If parameters not passed, build via config."""
```

Comment:
This can then be updated to explain how this is always initialized.

Reply:
This isn't intended to be always initialized via the configs, we would want to allow the user to build the SqlClient/Semantic models manually and pass it through as well.